### PR TITLE
Add CI gitactions job

### PR DESCRIPTION
**Pull request by RamLavi**
This PR introduces CI checks that will run on each PR

### CODE REVIEW

File: .github/workflows/main.yaml [hunk 1]
```diff
@@ -10,12 +10,12 @@ on:
   workflow_dispatch:
 
 jobs:
-  noop:
-    name: noop
+  lint:
+    name: lint
```
Comment (orelmisan) [inline .github/workflows/main.yaml:14]: nit:
```suggestion
    name: Lint
```

File: .github/workflows/main.yaml [hunk 2]
```diff
@@ -18,4 +18,11 @@ jobs:
...
         run: make lint
-
+  format:
+    name: fmt
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check out code
+        uses: actions/checkout@v3
+      - name: Run the format check
+        run: make fmt check-uncommitted
```
Comment (orelmisan) [inline .github/workflows/main.yaml:28]: You can configure the linter to catch this.
https://github.com/kiagnose/kiagnose/blob/main/.golangci.yml

**Review by orelmisan — COMMENTED** [review]
Thank you for the PR @RamLavi.
Please note that the [third](https://github.com/kiagnose/kubevirt-dpdk-checkup/pull/6/commits/64657b01fc10f5a321baee80f415adbc840d5924) commit has a wrong title.
Please see the inline comments.

File: .github/workflows/main.yaml (hunk 1 above)
Comment (RamLavi) [reply .github/workflows/main.yaml:14]: Done

File: .github/workflows/main.yaml (hunk 2 above)
Comment (RamLavi) [reply .github/workflows/main.yaml:28]: which linter specifically? I can't find it

Comment (orelmisan) [reply .github/workflows/main.yaml:28]: The linter will fail if the code is not properly formatted.

Comment (RamLavi) [reply .github/workflows/main.yaml:28]: Like! But adding this config doesn't suit this PR very well.
I will experiment adding this lint config in another PR and then remove this check. OK?

Comment (orelmisan) [reply .github/workflows/main.yaml:28]: > Like! But adding this config doesn't suit this PR very well. 

I agree

Comment (RamLavi) [reply .github/workflows/main.yaml:28]: Ok looks good enough. 
```
$ make lint
docker run --rm \
           --volume `pwd`:/home/ralavi/go/src/github.com/kiagnose/kubevirt-dpdk-checkup:Z \
           --workdir /home/ralavi/go/src/github.com/kiagnose/kubevirt-dpdk-checkup \
            docker.io/golangci/golangci-lint:v1.50.1 golangci-lint run --timeout 3m ./cmd/...
cmd/main.go:25: File is not `gofmt`-ed with `-s` (gofmt)
	fmt.Println("kubevirt-dpdk-checkup starting..."   )
make: *** [Makefile:45: lint] Error 1
```
will issue the PR

Comment (RamLavi) [reply .github/workflows/main.yaml:28]: https://github.com/kiagnose/kubevirt-dpdk-checkup/pull/8

Comment (RamLavi) [reply .github/workflows/main.yaml]: Removed the fmt job

**Review by orelmisan — APPROVED** [review]

**Pull request merged**